app/models/user.php

```diff
@@ -78,39 +107,77 @@ public static function get()
         return $row;
     }
 
+    /* CHECKERS FOR REGISTRATION */
+
+    //checks if firstname and lastname contain letters only
     public function nameChecker($name)
     {
         return ctype_alpha($name);
     }
 
+    //checks if password is same as password for confirmation
     public function passwordChecker()
     {
         return ($this->password == $this->confirm_password);
     }
 
+    //checks if username is not yet in use
     public function usernameChecker()
     {
         $db = DB::conn();
-        $is_username_existing = $db->row('SELECT username FROM user WHERE username = ?', array($this->username));
+        $is_username_existing = $db->row('SELECT username FROM user WHERE BINARY username = ? AND status=1', 
+            array($this->username));
 
         return (!$is_username_existing); //return true
     }
 
+    //checks if email is not yet in use
     public function emailChecker()
     {
         $db = DB::conn();
-        $is_email_existing = $db->row('SELECT email FROM user WHERE email = ?', array($this->email));
+        $is_email_existing = $db->row('SELECT email FROM user WHERE BINARY email = ? AND status=1', array($this->email));
 
         return (!$is_email_existing); //return true
     }
 
+    public function isNotBanned()
+    {
+        $db = DB::conn();
+
+        $params = array(
+            'username' => $this->username, 
+            'email'    => $this->email
+        );
+
+        $is_banned = $db->value('SELECT id FROM user WHERE username = :username AND status = 2 || 
+            email = :email AND status = 2', $params);
+
+        return (!$is_banned); //return true
```

Comment:
The return value and the comment is a bit misleading :sweat_smile: 
You can improve on the comment statement to make it easier to understand po :bow: